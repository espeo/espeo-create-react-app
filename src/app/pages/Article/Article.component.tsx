import React, { PureComponent } from 'react';
import dayjs from 'dayjs';

import {
  AppWrapper,
  TitleWrapper,
  Title,
  ContentWrapper,
  ArticleImage,
  ArticleBDate,
  ArticleBAuthor,
  ArticleBTitle,
  ArticleTitle,
  ArticleContent,
  ArticleBDescription,
} from '@core/components';

import { ArticleContainer, BackLink, BackLinkText, SourceLink } from './styles';

interface ArticleOwnProps {
  location: any;
  article: string;
}

type ArticleProps = ArticleOwnProps;

class Article extends PureComponent<ArticleProps> {
  render() {
    const { article } = this.props.location.state;

    return (
      <AppWrapper>
        <TitleWrapper>
          <Title text="Article Details" />
        </TitleWrapper>
        <ContentWrapper>
          <BackLink to="/">
            <BackLinkText>Back to Articles</BackLinkText>
          </BackLink>
          <ArticleContainer>
            <ArticleBTitle>{article.title}</ArticleBTitle>
            <ArticleBDescription>{article.description}</ArticleBDescription>
            <ArticleImage alt={article.title} src={article.urlToImage} />
            <ArticleBDate>
              {dayjs(article.publishedAt).format('YYYY, MMM DD ')}
            </ArticleBDate>
            <ArticleBAuthor>{article.author}</ArticleBAuthor>
            <ArticleTitle>{article.title}</ArticleTitle>
            <ArticleContent>{article.content}</ArticleContent>
            <SourceLink href={article.url} target="_blank">
              Read more here &gt; Source: {article.source.name}
            </SourceLink>
          </ArticleContainer>
        </ContentWrapper>
      </AppWrapper>
    );
  }
}

export default Article;
