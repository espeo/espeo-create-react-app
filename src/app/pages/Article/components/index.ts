export * from './ArticleContainer';
export * from './BackLink';
export * from './SourceLink';
