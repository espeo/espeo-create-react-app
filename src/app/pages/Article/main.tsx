import React, { PureComponent } from 'react';
import {
  AppWrapper,
  TitleWrapper,
  Title,
  ContentWrapper,
} from '@core/components';

interface ArticleOwnProps {
  location: any;
  article: string;
}

type ArticleProps = ArticleOwnProps;

class Article extends PureComponent<ArticleProps> {
  render() {
    const { article } = this.props.location.state;
    console.log(article);

    return (
      <AppWrapper>
        <TitleWrapper>
          <Title text="title" />
        </TitleWrapper>
        <ContentWrapper></ContentWrapper>
      </AppWrapper>
    );
  }
}

export default Article;
