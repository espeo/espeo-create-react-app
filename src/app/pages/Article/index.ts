import { connect } from 'react-redux';
import { compose } from 'redux';

import Article from './components/ArticleComponent/Article.component';

export default compose(connect())(Article);
