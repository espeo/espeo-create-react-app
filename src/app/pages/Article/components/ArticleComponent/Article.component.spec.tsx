import renderer from 'react-test-renderer';
import React from 'react';
import { compose } from 'redux';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import {
  withProviders,
  withAdditionalProperties,
  articleMock,
} from '@core/tests/theme-helpers';

import Article from './Article.component';

const locationMock = {
  state: {
    article: articleMock,
  },
};

const WrappedComponent = compose(
  withProviders,
  withAdditionalProperties({
    location: locationMock,
  }),
)(Article);

describe('Article component test suite', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <WrappedComponent />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should mount article with children', () => {
    const text = 'Article Details';
    const wrapper = mount(
      <BrowserRouter>
        <WrappedComponent />
      </BrowserRouter>,
    );
    expect(wrapper.find('h1').text()).toEqual(text);
  });

  it('should render article component', () => {
    const wrapper = mount(
      <BrowserRouter>
        <WrappedComponent />
      </BrowserRouter>,
    );
    expect(wrapper.find('.article')).toBeTruthy();
  });
});
