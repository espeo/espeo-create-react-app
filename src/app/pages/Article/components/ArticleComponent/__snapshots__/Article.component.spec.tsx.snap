// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test suite should match snapshot 1`] = `
<div
  className="sc-htpNat cTiQRV article"
>
  <div
    className="sc-ifAKCX kjOYKb"
  >
    <h1
      className="sc-jzJRlG gUwiRh"
    >
      Article Details
    </h1>
  </div>
  <div
    className="sc-bxivhb dKaubX"
  >
    <a
      className="sc-kAzzGY knAwDu"
      href="/"
      onClick={[Function]}
    >
      <p
        className="sc-chPdSV dmrJyt"
      >
        Back to Articles
      </p>
    </a>
    <div
      className="sc-cSHVUG heEdLu article-content"
    >
      <p
        className="sc-VigVT sc-jTzLTM lkNGIF"
      >
        Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
      </p>
      <p
        className="sc-bZQynM sc-gzVnrw fUMedX"
      >
        Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
      </p>
      <img
        alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
        className="sc-EHOje iXbLk"
        src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
      />
      <p
        className="sc-dnqmqq sc-iwsKbI iTtTSh"
      >
        2019, Nov 24 
      </p>
      <p
        className="sc-dnqmqq sc-gqjmRU jBFvQt"
      >
        Michael Wittershagen
      </p>
      <p
        className="sc-VigVT dqedxV"
      >
        Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
      </p>
      <p
        className="sc-bZQynM sc-htoDjs gWbtok"
      >
        Bis zu 25,6 Milliarden Dollar (etwa 23,11 Milliarden Euro) will Saudi-Arabien demnächst durch den Börsengang des staatlichen Ölkonzerns Aramco einnehmen. Dabei werden nur 1,5 Prozent der Unternehmensanteile verkauft, der Wert von Saudi Aramco wird damit auf e… [+1537 chars]
      </p>
      <a
        className="sc-kGXeez dyIyAp"
        href="https://www.faz.net/aktuell/sport/formel-1/motorsport/was-bedeutet-der-formel-e-start-in-saudi-arabien-16500882.html"
        rel="noopener"
        target="_blank"
      >
        Read more here &gt; Source: Faz.net
      </a>
    </div>
  </div>
</div>
`;
